# Benchmarking Alignment Strategies for Hi-C Reads in Metagenomic Hi-C Data
This repository contains the code used in the benchmark paper evaluating alignment strategies for Hi-C reads in metagenomic datasets.

## Description
We selected seven alignment strategies that have been used in Hi-C analyses: BWA MEM -5SP, BWA MEM default, BWA aln default, Bowtie2 default, Bowtie2 â€“very-sensitive-local, Minimap2 default, and Chromap default. We benchmarked them on one synthetic and seven real-world environments, and evaluated these tools based on the number of inter-contig Hi-C read pairs and their influence on downstream tasks, such as binning quality.

## Getting Started
1. **Clone the Repository** <br>
To obtain the necessary scripts, clone the repository using the following command:
```
git clone https://github.com/yuw053/MetaHi-C_Benchmark.git
```
<br>
1. Preprocessing <br>
Install BBTools by following the [BBTools installation guide](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/installation-guide/).

Once installed, run the preprocessing script:
```
chmod +x preprocessing.sh
./preprocessing.sh WGS_1_FILE_PATH WGS_2_FILE_PATH HIC_1_FILE_PATH HIC_2_FILE_PATH BBTOOLS_PATH OUTPUT_PATH
```
<br>
1. Assembly <br>
Create a Conda environment for the assembly and alignment steps:
```
conda env create -f sim3c.yml
conda activate sim3c
```
<br>

Run the assembly script:
```
chmod +x assembly.sh 
./assembly.sh WGS_CL_1_FILE_PATH WGS_CL_2_FILE_PATH OUTPUT_PATH
```
<br>
1. Alignment <br>
Run the alignment script:
```
chmod +x alignment.sh
./alignment.sh ASSEMBLY_FILE_PATH HIC_1_FILE_PATH HIC_2_FILE_PATH OUTPUT_PATH
```
<br>
1. MetaCC <br>
Install MetaCC by following the [MetaCC installation guide](https://github.com/dyxstat/MetaCC?tab=readme-ov-file#installation-guide).
<br>
For paired-end reads:
```
chmod +x downstream_pair.sh
./downstream_pair.sh ASSEMBLY_FILE_PATH ALINGMENT_FILE_PATH MODE OUTPUT_PATH
```
* MODE: Each mode corresponds to one alignment strategy. The following modes are available:
    * bwa_mem_pr_5sp
    * bwa_mem_pr_def
    * bwa_aln_sr
    * bowtie2_sr_def
    * bowtie2_sr_sensitive
    * minimap2_pr_def
    * chromap
<br>
For single-end reads:
```
chmod +x downstream_single.sh
./downstream_single.sh ASSEMBLY_FILE_PATH ALINGMENT_1_FILE_PATH ALINGMENT_2_FILE_PATH MODE OUTPUT_PATH
```
<br>
1. ImputeCC <br>
Install ImputeCC by following the [Impute installation guide](https://github.com/dyxstat/ImputeCC?tab=readme-ov-file#installation-guide).
<br>
Run:
```
chmod +x imputecc.sh
./imputecc.sh ASSEMBLY_FILE_PATH MODE OUTPUT_PATH
```
<br>
1. CheckM2 <br>
CheckM2 can be installed by following the instructions in the [CheckM2 GitHub repository](https://github.com/chklovski/CheckM2). <br>
To assess binning quality, you can use CheckM2 on the FINAL_BIN folder generated by ImputeCC:
```
checkm2 predict --threads 30 --input *.fa --output-directory checkm2_output
```
<br>
## License
This project is licensed under the MIT License - see the LICENSE.md file for details
